app/knight_and_items/knight.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from app.knight_and_items.weapon import Weapon
+from app.knight_and_items.armour import Armour
+from app.knight_and_items.potion import Potion
+
+
+class Knight:
+
+    def __init__(self, stats: dict) -> None:
+        self.name = stats.get("name")
+        self.hp = stats.get("hp")
+        self.power = stats.get("power")
+
+        self.armour = []
+        self.protection = 0
+
+        # wear armour
+        for new_armour_dict in stats.get("armour"):
```

Comment:
don't do many options in init. Use a single responsibility. Create separate functions for setting weapons, potions, and armor.